monkestation/code/modules/ranching/chickens/_chicken.dm

```diff
@@ -0,0 +1,458 @@
+		if(!(reagent in consumed_reagents))
+			consumed_reagents.Add(reagent)
+
+	if(!(given_item in consumed_food))
+		consumed_food.Add(given_item)
+
+	if(given_item.type in liked_foods && max_happiness_per_generation >= 10)
+		happiness += 10
+		max_happiness_per_generation -= 10
+
+	var/obj/item/food/placeholder_food_item = given_item
+	for(var/food_type in placeholder_food_item.foodtypes)
+		if(food_type in disliked_food_types)
+			happiness -= 10
+	if(user)
+		var/feedmsg = "[user] feeds [given_item] to [name]! [pick(feedMessages)]"
+		user.visible_message(feedmsg)
+
+	qdel(given_item)
+	eggs_left += rand(0, 2)
+	current_feed_amount ++
+	total_times_eaten ++
+
+/mob/living/simple_animal/chicken/Hear(message, atom/movable/speaker, message_langs, raw_message, radio_freq, spans, list/message_mods = list())
+	. = ..()
+	if(speaker != src && !radio_freq && !stat)
+		if (speaker in Friends)
+			speech_buffer = list()
+			speech_buffer += speaker
+			speech_buffer += lowertext(html_decode(message))
+
+/mob/living/simple_animal/chicken/proc/handle_speech()
+	if (speech_buffer.len > 0)
+		var/who = speech_buffer[1] // Who said it?
+		var/phrase = speech_buffer[2] // What did they say?
+		if (findtext(phrase, "chickens")) // Talking to us
+			if(findtext(phrase, "follow"))
+				if (ai_controller.blackboard[BB_CHICKEN_CURRENT_LEADER])
+					if(Friends[who] > Friends[ai_controller.blackboard[BB_CHICKEN_CURRENT_LEADER]]) // following you bby
+						ai_controller.blackboard[BB_CHICKEN_CURRENT_LEADER] = who
+						ai_controller.queue_behavior(/datum/ai_behavior/follow_leader)
+				else
+					if (Friends[who] >= CHICKEN_FRIENDSHIP_FOLLOW)
+						ai_controller.blackboard[BB_CHICKEN_CURRENT_LEADER] = who
+						ai_controller.queue_behavior(/datum/ai_behavior/follow_leader)
+
+			else if (findtext(phrase, "stop"))
+				ai_controller.blackboard[BB_CHICKEN_CURRENT_ATTACK_TARGET] = null
+
+			else if (findtext(phrase, "stay"))
+				if(ai_controller.blackboard[BB_CHICKEN_CURRENT_LEADER] == who)
+					AIStatus = AI_STATUS_ON
+					ai_controller.blackboard[BB_CHICKEN_CURRENT_LEADER] = null
+					SSmove_manager.stop_looping(src)
+
+			else if (findtext(phrase, "attack"))
+				if (Friends[who] >= CHICKEN_FRIENDSHIP_ATTACK)
+					for (var/mob/living/target in view(7,src)-list(src,who))
+						if (findtext(phrase, lowertext(target.name)))
+							if (istype(target, /mob/living/simple_animal/chicken))
+								return
+							else if((!Friends[target] || Friends[target] < 1))
+								if(ai_controller.blackboard[BB_CHICKEN_CURRENT_LEADER])
+									ai_controller.blackboard[BB_CHICKEN_CURRENT_LEADER] = null
+								ai_controller.blackboard[BB_CHICKEN_CURRENT_ATTACK_TARGET] = target
+						break
+		speech_buffer = list()
+
+/mob/living/simple_animal/chicken/Life()
+	. =..()
+	if(!.)
+		return
+
+	handle_speech()
+
+	if(COOLDOWN_FINISHED(src, age_cooldown))
+		COOLDOWN_START(src, age_cooldown, age_speed)
+		age ++
+
+	if(age > 100)
+		src.death()
```

Comment:
Might be worth getting rid of this, because slimes don't die that quickly

Reply:
you can't get dream chickens without one dying from old age